### ISSUE TITLE
Kind plea from visually impaired advanced userDear specialists,

### ISSUE THREAD

**Issue by JanuszChmiel** [XusinboyBekchanov/VisualFBEditor#5]

**Comment by JanuszChmiel** [comment]
Dear specialist, I do not see at all and when I have analysed yours Examples I think, that I could write apps even without need to specify object position values. What will happen, if I do not specify object borders and other number related object information? You have even included altop, alleft and other adjectives to specify object position too. Can I use those adjectives instead of Object position values in Pixels? Or I will occur app crashes and system unstability?

**Comment by XusinboyBekchanov** [comment]
You can, only a minimum you must specify the size of the controls.

**Comment by JanuszChmiel** [comment]
So if I will ommit values numbers app will crash or it will look very ugly?

**Comment by XusinboyBekchanov** [comment]
It turns out there will be no problems. The sizes are supplied with default values.

**Comment by JanuszChmiel** [comment]
Oh its perfect. I love yours Free Basic GUi unit because it is very close to
FNX Basic commands for GUI creation. Object programming.
Which version of Free Basic compiler do I have to use to compile yours units
reliably?

**Comment by JanuszChmiel** [comment]
aaaaI have one kind suggestion to you.
Try to debate with Free Basic compiler developers if it is necessary to
obtain The permission from them so you could include compatible Free Basic
compiler with yours .7z archive with yours IDE.
Thank you for yours complex work and I will be able to develop reliable and
fast apps without pseudocompilers as FNX Basic. Because Free Basic compile
directly to machine code.

**Comment by XusinboyBekchanov** [comment]
The IDE itself is compiled using version 1.06.0. But the programs being created are compiled with new versions of the compiler. Only in new versions, when opening files through Open Close, you need to use fixed WString * 1024, dynamic WString Ptr does not work. I am preparing a new version of the IDE, it will be compiled in new compilers. Regarding the inclusion of the compiler in zip: I think that in each new version of the IDE each time you have to download the compiler and this increases the size of the IDE bundle. I want the IDE kit to be minimally sized.

**Comment by JanuszChmiel** [comment]
May I use yours units for GUI creation in my freeware apps and can I
distribute those apps for free by using Internet?
I would have The zero chance to create something so perfect like you have
created. I have spent many many hours while I have searched for GUI creation
modules for Free Basic But The most of those module have contained subs and
functions which have used object position values as mandatory value. SO I
could not use them. Developer with no sight at all can not correctly
determine The values. YOurs units are perfect because It is possible to use
adjectives for positioning object left, right, ETC. Similarly like BOrland
Delphi VCL special GUi creation unit allow.
This is The reason why I Am not sad, that The size of compiled executables
starts on 1 MB. Because I prefer bigger .exe size than having pain while
learning lov level WIN API calls.
Thank you that you have decided even to use non GTK variant for Windows. Or
did you included special bridge between GTk and MSAA? Because I have found
big gtk module in The distribution. With The ide in the same folder.
Or WIndows api calls are used without GTK?
GTK was big trouble for screen readers. Its Windows compiled variant do not
contain software bridge which allow screen readers to use The GUI.
So really very well done. I will setup The compiler and I will start
developing. The last thank is for you because all GUI elements are fully
accessible with screen readers. The reaction speed between key press and
screen reader output while I have tested groupbox object was outstanding.
YOu have implemented The GUI toolkit really perfectly. No Windows crashes.
You had to spent many many hours on it so I deeply appreciate you like a
human beeing and as a professional programmer.
Yours GUI toolkit will allow visually impaired programmers to use Free Basic
compiler.
